web/pandas_web.py

```diff
@@ -442,16 +439,16 @@ def main(
     before copying them. ``.md`` files are transformed to HTML.
     """
     # Sanity check: validate that versions.json is valid JSON
-    versions_path = os.path.join(source_path, "versions.json")
-    with open(versions_path, encoding="utf-8") as f:
+    versions_path = pathlib.Path(source_path) / "versions.json"
+    with versions_path.open(encoding="utf-8") as f:
         try:
             json.load(f)
         except json.JSONDecodeError as e:
             raise RuntimeError(
                 f"Invalid versions.json: {e}. Ensure it is valid JSON."
             ) from e
 
-    config_fname = os.path.join(source_path, "config.yml")
+    config_fname = pathlib.Path(source_path) / "config.yml"
```

Comment:
You are making source_path a pathlib twice. Ideally we want paths to be a pathlib since they are created. Not create them as str and convert when there is an operation. For everything that is a path, also config_path and others